tests/test_geo.py

```
@@ -0,0 +1,88 @@
+    [(91.0, 0.0), (-91.0, 0.0), (0.0, 181.0), (0.0, -181.0)],
+)
+def test_coordinate_rejects_out_of_range(lat: float, lon: float) -> None:
+    with pytest.raises(ValueError):
+        Coordinate(latitude=lat, longitude=lon)
+
+
+def test_coordinate_is_hashable_and_frozen() -> None:
+    """Coordinates are used as dict keys in the incremental territory update."""
+    coord = Coordinate(latitude=1.0, longitude=2.0)
+    assert {coord: "ok"}[coord] == "ok"
+    with pytest.raises(Exception):
```

Ruff (0.16.0)

[warning] Do not assert blind exception: `Exception`

(B017)